### PR TITLE
Confirmation button

### PR DESCRIPTION
**Pull request by abbiecoghlan**
Part one of [#297](https://github.com/FlourishHealth/flourish/issues/297) 

Tested on Web, Android, and iOs

Default:
Screen Shot 2022-10-18 at 2 55 32 PM

Custom:
Screen Shot 2022-10-18 at 2 55 39 PM

https://user-images.githubusercontent.com/66394682/196531893-58215f13-d574-4268-b2e1-1357f8bc8e5d.mov

### CODE REVIEW

File: packages/ui/src/Button.tsx [hunk 1]
```diff
@@ -92,71 +100,101 @@ export function Button({
...
+          flexDirection: "row",
+          paddingHorizontal: 4 * 2,
+        }}
+        onPress={debounce(
+          async () => {
+            Unifier.utils.haptic();
+            setLoading(true);
+            try {
+              if (withConfirmation) {
+                if (!showConfirmation) {
```
Comment (swanka) [inline packages/ui/src/Button.tsx:152]: You can combine this conditional with the one above: `withConfirmation && !showConfirmation`

File: packages/ui-demo/src/Button.stories.tsx [hunk 2]
```diff
@@ -8,28 +8,28 @@ const allColorButtons = (props: Partial<ButtonProps>) => {
   return (
     <StorybookContainer>
       <Box paddingY={1}>
-        <Button text="Default" onClick={action("clicked")} {...props} />
+        <Button text="Default" onClick={() => action("clicked")} {...props} />
```
Comment (joshgachnang) [inline packages/ui-demo/src/Button.stories.tsx:11]: Ah ha, good catch!

File: packages/ui/src/Button.tsx [hunk 3]
```diff
@@ -92,71 +100,101 @@ export function Button({
...
-        </Box>
-      )}
-    </TouchableOpacity>
+            skipLinking
+            weight="bold"
+          >
+            {text}
+          </Text>
+        )}
+        {(loading || loading) && (
```
Comment (joshgachnang) [inline packages/ui/src/Button.tsx:191]: I know this isn't your code, but this could be just `loading` or `Boolean(loading)` for extra safety.

**Review by joshgachnang — APPROVED** [review]

File: packages/ui/src/Button.tsx (hunk 1 above)
Comment (abbiecoghlan) [reply packages/ui/src/Button.tsx:152]: Ah good catch!!

**Pull request merged**